src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java

```diff
@@ -18,131 +18,49 @@
     public static final String DEFAULT_DISPLAY_NAME = "Instructor";
 
     // Note: be careful when changing these variables as their names are used in *.json files.
+
+    /** Required fields. */
     public String googleId;
     public String courseId;
     public String name;
     public String email;
-    public Boolean isArchived;
+
+    /** Optional fields. */
     public String key;
     public String role;
-    public boolean isDisplayedToStudents;
     public String displayedName;
-
+    public Boolean isArchived;
+    public boolean isDisplayedToStudents;
     public InstructorPrivileges privileges;
 
-    /**
-     * Creates a new instructor with default access level and default displayedName.
-     *
-     * @deprecated only to be used for testing
-     */
-    @Deprecated
-    public InstructorAttributes(String googleId, String courseId, String name, String email) {
```

Comment:
I can add them to builder constructor, but can we consider to use this fields as optional? For, example if displayedName is not provided we can use name param which is required as default value. And for role specify INSTRUCTOR_PERMISSION_ROLE_COOWNER as default for example.